### PR TITLE
KVS Migration - Screen Sweep Tables from Validation

### LINKED ISSUE #2137: KVS migration fails in the validation stage for sweep.priority table
This causes some confusion when running the migration. The failure occurs as kvs migration writes to the `sweep.priority` in the `fromDatabase` but not in the `toDatabase`. This is not an issue in the migration stage but in the validation stage. One solution will be to bypass sweep tables when validating.

Ideally, we dont want to migrate the `sweep` namespace at all.

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**: Fix #2137 

**Implementation Description (bullets)**:
- Remove tables from the Sweep namespace from validation.
- Add a bunch of tests for the validator.

**Concerns (what feedback would you like?)**:
- Is the condition too strong? Should I be removing strictly priority and progress?
- Do we need to support unvalidatable tables?

**Where should we start reviewing?**: `KeyValueServiceValidators.java`

**Priority (whenever / two weeks / yesterday)**: whenever?

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2244)

### CODE REVIEW

File: docs/source/release_notes/release-notes.rst [hunk 1]
```diff
@@ -44,8 +44,14 @@ develop
     *    - Type
          - Change
 
+    *    - |fixed|
+         - KVS migrations will no longer verify equality between the from and to KVSes for the sweep priority and progress tables.
+           Note that these tables are still *migrated* across, as they provide heuristics for timely sweeping of tables.
+           However, these tables are expected to change as the migration process performs writes to the to-KVS.
```
Comment (hsaraogi) [inline docs/source/release_notes/release-notes.rst]: I think the `from-kvs` is being swept and hence `sweep.priority` there is being updated. Theoretically, there should be nothing to sweep in the `to-kvs` as there is only the latest version of the data and migrations are offline.

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidators.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+
+public final class KeyValueServiceValidators {
+    private KeyValueServiceValidators() {
+        // utility
+    }
+
+    /**
+     * Returns tables that need to be validated.
+     * Generally speaking, this excludes tables that are not controlled by the transaction table, as well as
```
Comment (hsaraogi) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidators.java]: Not sure I understand this statement :/. Tables not controlled by the transaction table?

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidators.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+     */
+    public static Set<TableReference> getValidatableTableNames(
+            KeyValueService kvs,
+            Set<TableReference> unmigratableTables) {
+        Set<TableReference> tableNames = KeyValueServiceMigrators.getMigratableTableNames(kvs, unmigratableTables);
+        return removeSweepTableReferences(tableNames);
+    }
+
+    /**
+     * Removes any table references that are part of Sweep, defined as being a part of the Sweep namespace.
```
Comment (hsaraogi) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidators.java]: The comment seems redundant as it says what the code does.

Comment (jeremyk-91) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidators.java]: That's fair. I wrote this with the intention to document that "Sweep namespace = part of Sweep", though that's probably a reasonable assumption to make 😄

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidators.java (hunk 2 above)
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidators.java]: Non-transactional tables, I believe. This was copied from `getMigratableTableNames` actually

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/schema/KeyValueServiceValidatorsTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.impl.InMemoryKeyValueService;
+import com.palantir.atlasdb.transaction.impl.TransactionConstants;
+
+public class KeyValueServiceValidatorsTest {
+    private static final TableReference SWEEP_PRIORITY = TableReference.createFromFullyQualifiedName("sweep.priority");
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/schema/KeyValueServiceValidatorsTest.java]: Yep, agree that this makes sense.

File: docs/source/release_notes/release-notes.rst (hunk 1 above)
Comment (jeremyk-91) [inline docs/source/release_notes/release-notes.rst]: Yes, the `from-kvs` could be swept.

Regarding the `to-kvs`, there won't be anything to sweep, but my concern was that the write counts in the `to-kvs` would be affected by writing?

**Pull request merged**